src/ray/gcs/gcs_server/gcs_actor_scheduler.cc

```diff
@@ -311,15 +312,27 @@ void GcsActorScheduler::HandleWorkerLeaseGrantedReply(
   }
 }
 
-void GcsActorScheduler::OnRuntimeEnvSetupFailure(std::shared_ptr<GcsActor> actor,
-                                                 const NodeID &node_id) {
+void GcsActorScheduler::HandleRequestWorkerLeaseCanceled(
+    std::shared_ptr<GcsActor> actor, const NodeID &node_id,
+    rpc::RequestWorkerLeaseReply::CancelType cancel_type) {
   RAY_LOG(ERROR)
-      << "Failed to lease worker from node " << node_id << " for actor "
+      << "The lease worker request from node " << node_id << " for actor "
       << actor->GetActorID() << "("
       << actor->GetCreationTaskSpecification().FunctionDescriptor()->CallString() << ")"
-      << " as the runtime environment setup failed, job id = "
-      << actor->GetActorID().JobId();
-  schedule_failure_handler_(actor, /*runtime_env_setup_failed=*/true);
+      << " has been canceled, job id = " << actor->GetActorID().JobId()
+      << ", cancel type: " << rpc::RequestWorkerLeaseReply::CancelType_Name(cancel_type);
+
+  switch (cancel_type) {
+  case rpc::RequestWorkerLeaseReply::PLACEMENT_GROUP_REMOVED:
+    schedule_failure_handler_(actor, ActorSchedulingFailedType::PLACEMENT_GROUP_REMOVED);
```

Comment:
Do we need a separate enum? Why don't we pass rpc::RequestWorkerLeaseReply::PLACEMENT_GROUP_REMOVED: directly?

Reply:
because it's two different types,  `WorkerLeaseCanceledType` vs `ActorSchedulingFailedType`, actually I think `WorkerLeaseCanceled` belongs to one of `ActorSchedulingFailedTypes` as  `WorkerLeaseCanceled` will only be generated when the request worker lease has been canceled and `ActorSchedulingFailed` will also be generated when there is no resource to schedule actors in GCS scheduling.

Reply:
What about this; 

```
/// \param cancel_type The reason why scheduling has been cancelled. 
  void OnActorSchedulingFailed(std::shared_ptr<GcsActor> actor,
                               const CancelType cancel_type == rpc::CancelType::NotCancelled);
if (cancel_type == NOT_CANCELLED) {
    // We will attempt to schedule this actor once an eligible node is
    // registered.
    pending_actors_.emplace_back(std::move(actor));
    return;
  } else if (cancel_type == PlacementGroupRemoved) {
      blah blah
   }
}
```

And in the docstring we can explain CancelType

Reply:
I think with this change, we don't need redundant conversion https://github.com/ray-project/ray/pull/20474#discussion_r767364609?

Reply:
updated.